### ISSUE TITLE
Create mobile version of the Guide Pages and ensure responsiveness

### ISSUE THREAD

**Issue by abenipa3** [hackforla/website#1515]
### Dependency
- [x] #2978
- [ ] #3170 
- [x] ~Mobile navigation audit. #2190~
- [x] https://github.com/hackforla/website/issues/1630

### Overview
We learned that Jekyll is a feasible option because it can render Markdown files into HTML as seen in #1085. 

Now we need to create a mobile version of it and ensure that the codes are responsive on all screens. 

### Action Items

- [x] Create a mobile version in the prototype.
- [x] Design mobile version of guide pages.
- [ ] Consult with the product and design team.

### Future Items

- [ ] Implement prototype into the HfLA repository. (The prototype is currently in @alyssabenipayo for testing purposes.)
- [ ] Create documentation of how the Jekyll Guide Pages works and how the team can use it.
- [ ] Add it HfLA's Drive. Get approval from the team.

### Resources/Instructions
[Guide Pages Figma File](https://www.figma.com/file/0RRPy1Ph7HafI3qOITg0Mr/Hack-for-LA-Website?node-id=11036%3A91034)
[HfLA Guide Pages Design Issue](https://github.com/hackforla/website/issues/1525)
[Alyssa's Repository for the Guide Pages Prototype](https://github.com/alyssabenipayo/hello-world)

**Comment by abenipa3** [comment]
Progress: Just assigned this issue to myself.
Blocks: N/A
Availability: Will be available starting on 5/13 to work on this issue.
ETA: I might have some progress on 5/15.

**Comment by qiqicodes** [comment]
@alyssabenipayo Can you let us know your progress in regards to this issue? Thank you =]

Progress:
Blockers:

**Comment by abenipa3** [comment]
Progress: <STRIKE>Planning to work on it today (5/15). Will update this post before our Sunday meeting.</STRIKE> Started working on creating the mobile version with CSS. Experimenting between hiding the navbar and creating an accordion. 

<details><summary>Screenshot for mobile version without navbar</summary>

![image](https://user-images.githubusercontent.com/38295612/118406687-b61def80-b631-11eb-8071-afb657ac8343.png)

</details>

Blocks: Have been focusing on Hackathon this past week. Learned a lot from the experience (especially with CSS and a bit of JavaScript) that I’m excited to continue with this issue.

Availability: Open this coming week. (Yay!)

ETA: 5/22.

**Comment by abenipa3** [comment]
Progress:

- Created wireframes for the mobile designs as mentioned in #1525.
- Applied the media query works in the prototype to hide nav-bar in all mobile devices as seen below. [Repo here](https://github.com/alyssabenipayo/guidepages-prototype).

![githubissue](https://user-images.githubusercontent.com/38295612/119271047-9ac45e80-bbb4-11eb-8887-9a3d59888664.gif)

Blocks: N/A

Availability: Open!

ETA: 5/23/21

**Comment by abenipa3** [comment]
Progress: 

- Reviewed wireframes with Design and Product as stated in #1525 
- Next Objective:
   - Add code with a collapsible menu to the prototype.

Block: N/A
Availability: Open
ETA: 5/29/21

**Comment by abenipa3** [comment]
Feedback from Dev Team: We learned that the most feasible way to develop is to hide the nav-bar. 

Will have the prototype ready by Sunday.

**Comment by abenipa3** [comment]
Progress:

Added page-card and media query to hide the nav-bar as seen below.

<details><summary>Screen capture for page-card update</summary>

![image](https://user-images.githubusercontent.com/38295612/120091624-c1f1c300-c0c1-11eb-89ac-8f4afeabe4f9.png)

</details>

<details><summary>Screen capture for Media Query Update</summary>

![prototype-guide-page](https://user-images.githubusercontent.com/38295612/120093245-d6d45380-c0cd-11eb-897a-90193dc08268.gif)

</details>


Block: N/A

Availability: Open

ETA: Ready for review.

**Comment by abenipa3** [comment]
Feedback from Sunday Meeting: We would like to update the mobile prototype where the table of contents is hiding at the top (hamburger menu) while a top scroll button appears after the user scrolls through the page.


Note for future references (potential new issue): Create an accordion for mobile.

**Comment by abenipa3** [comment]
Progress: Added scroll to top button for mobile.
<details><summary>Screen capture</summary>

![scrolltop](https://user-images.githubusercontent.com/38295612/120914126-cdf4fc00-c650-11eb-9c3a-39e57514721f.gif)


</details>

Block: Will review new wireframes with design team.

Availability: Open

ETA: TBD

**Comment by abenipa3** [comment]
Progress: Scroll to top button bug fixed! Also added links to the page card, which will only appear on mobile.

![scrolltopandlinks](https://user-images.githubusercontent.com/38295612/121794099-7f001700-cbba-11eb-8cad-6ea3aa2469d7.gif)


Would like feedback on the links on page card as seen in the GIF above. 
Had to create a div in markdown file for it to work. Any suggestions?
![image](https://user-images.githubusercontent.com/38295612/121817286-494c4400-cc35-11eb-9f1e-d94d990855f6.png)

Block: None. 

Availability: Open

ETA: 6/13/21

**Comment by abenipa3** [comment]
Feedback from 6/13 Meeting:

- We want to avoid having to add HTML in the markdown so Akib suggested toc.
    - Will revisit the toc prototype to test it!
- ETA: Latest 6/19. Earliest 6/15.

**Comment by abenipa3** [comment]
Updated ETA: 6/19.

**Comment by Sihemgourou** [comment]
@alyssabenipayo We are putting this back into in progress. Tell us if you need some review :)

**Comment by abenipa3** [comment]
Progress: We finalized the design on Thursday with the Team. Currently implementing it to the prototype.
![image](https://user-images.githubusercontent.com/38295612/122664929-2bad3c00-d159-11eb-82d5-6fbebf70ec0e.png)

Block: None.

Availability: Open

ETA: 6/26/21

**Comment by abenipa3** [comment]
Progress: (Update for Tuesday's Meeting) Still currently implementing the design into the prototype. 
![image](https://user-images.githubusercontent.com/38295612/123024522-081f0700-d38e-11eb-82ce-b5c9bb46c5a2.png)
Blocker: Will be busy till Thursday. Preparing for an interview.
Availability: Open
ETA: 6/26/21

**Comment by abenipa3** [comment]
Progress: 

- Added "Overview" in the front matter of the MD files. 
- Included nav links under "In This Guide" into the md files. (Will explain in the dev meeting.)
- Fixed and updated hero section. Please see Prototype Preview below.
- Applied media query to page cards and hero section.

<details><summary>Prototype Preview (Desktop)</summary>

![image](https://user-images.githubusercontent.com/38295612/123540570-8f54dd80-d6f4-11eb-8748-de35e65600c7.png)


</details>

<details><summary>Prototype Preview (Mobile)</summary>

![image](https://user-images.githubusercontent.com/38295612/123541383-26bc2f80-d6f9-11eb-93f1-e05f31947529.png)


</details>



Blocker: (Notes) Minor bugs encountered as seen in the previews below. Will need to adjust the styling before it's ready to go. Plus, will fix responsiveness for Guide Title and Page Cards. 

<details><summary>Parallelogram in the Hero Section is larger than the hero image </summary>

![image](https://user-images.githubusercontent.com/38295612/123540617-ce832e80-d6f4-11eb-90a3-36b5537e3bd4.png)

</details>


Availability: Open

ETA: 6/30/21

Note: The pull request below was from my own repo. Please feel free to ignore it.

**Comment by abenipa3** [comment]
Progress: Looked at it but haven't done much 'cause I was not feeling well this past week.
Blocker: Same as previous post. Will pick it up again today.
Availability: Open
ETA: 7/6/21

**Comment by abenipa3** [comment]
Progress: Fixed parallelogram so that it maintains its size when testing on different screen sizes.
Blockers: Need to make the rest of the page responsive. Also finding a cleaner way to make it responsive.
Availability: Open
ETA: 7/16/21

**Comment by abenipa3** [comment]
**Progress:**

![guidepages-071221](https://user-images.githubusercontent.com/38295612/125380180-a297b800-e346-11eb-8e7c-c148ab5dbe72.gif)

- Hero Section Text, Diagonal, and Image are now responsive.
   - Able to make text and diagonal stay in one place when shown on smaller screens.
- Updated guide-pages.html layout to have "In this Guide" Links show on mobile-only without using divs in markdown.
- Added Preview and Overview to MD's front matter. (To be discussed with Dev Team)
<details><summary>Preview and Overview in Front Matter</summary>

![image](https://user-images.githubusercontent.com/38295612/125380425-08843f80-e347-11eb-8b7d-bf77cbab46bc.png)


</details>

- Added a new page card to show Overview.




**Blockers:** 

TLDR; Front matter inquiry and wrapping up on responsiveness. Other feedback welcome as well!

- How to add links to the front matter?
  - Tried to google how to do so because some overviews contain external links. It would be nice to keep the overview in the front matter since it's contained in a div (guidepages.html layout) to be able to hide it on mobile.

- Should I start implementing this prototype into the HfLA repo?

- Would like feedback on overall progress as well.  

**Availability:** Open

**ETA:** 7/16/21

**Comment by abenipa3** [comment]
Progress: Updated Header and Style for Desktop version as seen below. Question about margins and grid containers. Also will continue to work on styling before our next meeting.
![image](https://user-images.githubusercontent.com/38295612/126031437-7790fcec-5972-47f0-9038-cbdd4cdb4e95.png)

Blocker: N/A

Availability: Open

ETA: 7/20/21

Note: Will meet with Josh on Monday (7/19) to review progress and wrap up this issue.

**Comment by abenipa3** [comment]
@Sihemgourou @ExperimentsInHonesty Hello! I've completed creating responsive Guide Pages. Please see the following link from my repo and feel free to let me know your thoughts: https://alyssabenipayo.github.io/guidepages-prototype/

Note: Please see these following links:
![image](https://user-images.githubusercontent.com/38295612/126239636-78c0e5fd-6eeb-48f3-9165-bb7bd0b5d631.png)

**Comment by abenipa3** [comment]
Progress: Changed css classes and link's font colors to reflect HfLA's Design System.
![image](https://user-images.githubusercontent.com/38295612/126852641-3d4d4fc1-45a3-4265-aff2-6989bb6986e4.png)

Blocker: Figuring out how to resolve responsiveness bug as seen below. Last week, we thought we found a solution by adding a margin-left on the sticky nav, but it doesn't apply when scrolling down the second-page card. 

![pagecards](https://user-images.githubusercontent.com/38295612/126909377-59702d62-6082-4210-91ea-f7378df09f59.gif)

So far tried display: grid and display:flex, but no working solution yet. Will bring up this issue again on Sunday.

Availability: Open

ETA: 7/26/21

**Comment by abenipa3** [comment]
Using this space to make notes of what to do next before the new ETA (7/30/21):

Progress as of 7/27/21: 
Responsiveness bug on page cards resolved!

Blocker: 
Discovered another responsiveness bug for screens (769px — 1024px) as seen below.
![image](https://user-images.githubusercontent.com/38295612/127295445-9ff0d264-ad23-4205-9324-c01434b38031.png)

Need to fix the alignment for mobile devices as seen below, too. (I love CSS. 😂)
![image](https://user-images.githubusercontent.com/38295612/127295713-73ffdf7b-f611-42f0-aad9-910dc9817803.png)

Availability: Open

**Comment by abenipa3** [comment]
Progress: 

- Fixed alignment for page cards on mobile and desktop. 
- The only concern is that the parallelogram in the hero section. 
- Other than that, this prototype is ready to review.

Blockers: None. Would like to gather feedback from our dev team before consulting with product and design.

Availability: Open

ETA: ~Tonight at 11:59pm (Will update more later.)~ Pending.

Preview: https://alyssabenipayo.github.io/guidepages-prototype/

**Comment by abenipa3** [comment]
Just found out how it really looks like on mobile. 

<details><summary>Current Mobile Preview</summary>

![image](https://user-images.githubusercontent.com/38295612/127780722-8c2ebbb4-86c4-4f08-9a9a-41eeef51d124.png)



</details>


I manually coded the parallelogram. I think a quick solution to this is downloading the svg of the hero image with the shape from our Figma and implement it on our prototype. With that, we would avoid visual issues in the future. 

~**Going to reach out to Design and Dev teams  for feedback and update again today.**~

Updated ETA: 8/1/21

**References:**

<details><summary>Original Image without Parallelogram</summary>

![Mask Group](https://user-images.githubusercontent.com/38295612/127780716-ae48ccde-2086-4cf8-ba5f-15a6599b14d8.png)


</details>

<details><summary>Original Image with Parallelogram</summary>

![Getting-the-most](https://user-images.githubusercontent.com/38295612/127780712-a4ccea04-726a-43fa-9f3b-2b58409ec4c9.png)


</details>


**UPDATE (8/5/21): Design and Dev teams said that downloading the svg image with the parallelogram is a workable solution. Will proceed to have this issue reviewed with Product.**

**Comment by abenipa3** [comment]
@Sihemgourou @ExperimentsInHonesty Hello! Guide Pages Prototype is ready to review. I've consulted with our dev and design teams to see if they have feedback, and they said it looks good so far.

Please see the link for the prototype below and feel free to let me know if you have any questions! Thanks.

https://alyssabenipayo.github.io/guidepages-prototype/

**Comment by Sihemgourou** [comment]
Great @alyssabenipayo ! could you put it in the review column please :)

**Comment by abenipa3** [comment]
Update from 8/15/2021's Meeting:

- [ ]  Make the sample issues into actual issues. (We want to be able to tell that it works when it's a real issue.)
- [x] Add submenus to the 2FA's FAQ. 
- [ ] Update the page cards so that there's padding. Refer to HfLA's About Section as a reference.
- [x] Replicate the full guide of [How to Make a Guide](https://docs.google.com/document/d/1-8gmeC-wnfM8C8fVvmTP1BLA2WfNjKH_XJrHB1OAV6Q/edit#).
- [ ] Continue working on Markdown converter for guides ([10](https://github.com/hackforla/guides/issues/10)).

Blockers: None.

Availability: 8-10 hours

ETA: 8/21/21

Preview: https://alyssabenipayo.github.io/guidepages-prototype/

**Comment by abenipa3** [comment]
Additional Notes from Previous Update:
- [ ] Check the headings for each section title.
- [ ] Styling not applied to subsections of the navbar in mobile and tablet. Need to update.
- [x] Edit the indentation of the subsections (See JavaScript).

**Comment by abenipa3** [comment]
# Update 8/20/21

Will be redesigning the Guide Pages as mention in #1630. Status of this issue: TBD

**Comment by abenipa3** [comment]
Issue Update 8/27/21:

- Currently redesigning the Guide Pages.
- This issue now has a dependency in the Mobile Navigation Audit (#2190).
- Updated issue description with a _Dependency_ section, added Dependency Label, and moved to the ice box.
